### PR TITLE
luci-app-firewall: add masq6 option for zones

### PR DESCRIPTION
**Pull request by 1715173329**
Allow configure Masquerading6 via LuCI interface.

### CODE REVIEW

File: applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js [hunk 1]
```diff
@@ -157,18 +157,31 @@ return view.extend({
...
-			_('Enable network address and port translation (NAT or NAPT) for outbound traffic on this zone. This is typically enabled on the <em>wan</em> zone.'));
+		o = s.taboption('general', form.Flag, 'masq', _('IPv4 Masquerading'),
+			_('Enable network address and port translation IPv4 (NAT4 or NAPT4) for outbound traffic on this zone. This is typically enabled on the <em>wan</em> zone.'));
 		o.editable = true;
 		o.tooltip = function(section_id) {
+			var family = uci.get('firewall', section_id, 'family')
 			var masq_src = uci.get('firewall', section_id, 'masq_src')
 			var masq_dest = uci.get('firewall', section_id, 'masq_dest')
-			if (masq_src || masq_dest)
+			if ((!family || family === 'ipv4') && (masq_src || masq_dest))
```
Comment (jow-) [inline applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js:167]: The check is overly strict, `family` may be `inet`, `inet6`, `ipv4`, `ipv6`, `4`, `6`. I suggest to do `family.indexOf('6') == -1`

Comment (1715173329) [reply applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js:167]: Got it, thanks!

File: applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js [hunk 2]
```diff
@@ -157,18 +157,31 @@ return view.extend({
...
 			var masq_src = uci.get('firewall', section_id, 'masq_src')
 			var masq_dest = uci.get('firewall', section_id, 'masq_dest')
-			if (masq_src || masq_dest)
+			if ((!family || family.indexOf('6') == -1) && (masq_src || masq_dest))
 				return _('Limited masquerading enabled');
 
 			return null;
 		};
 
+		o = s.taboption('general', form.Flag, 'masq6', _('IPv6 Masquerading'),
```
[the remark this answers is not in the document]
Comment (1715173329) [reply applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js:173]: Sorry I don't even notice there's no masq6 support in fw3.. Should I also check firewall version for masq_src / masq_dest like this?

```js
if (!fw4) {
	o.depends('family', '');
	o.depends('family', 'ipv4');
	o.datatype = 'list(neg(or(uciname,hostname,ipmask4)))';
} else
	o.datatype = 'list(neg(or(uciname,hostname,ipmask)))';
```

**Comment by jow-** [comment]
And one final remark; I am not sure if we should expose the `masq6` option this prominently. At the very least it should be hidden from the tabular zone overview (by removing `o.editable = true;`) and maybe it should be even moved into the advanced section.

IPv6 masquerading on a zone should be the exception (as last resort after PD and relay mode) rather than the norm.

**Comment by 1715173329** [comment]
> And one final remark; I am not sure if we should expose the masq6 option this prominently. At the very least it should be hidden from the tabular zone overview (by removing o.editable = true;) and maybe it should be even moved into the advanced section.

Updated and moved it into the advanced section.

**Comment by jow-** [comment]
Merged, thanks!

**Pull request merged**